poolmgr/gp.go

```diff
@@ -327,9 +358,46 @@ func (gp *GenericPool) createFunctionDeployment(metadata *fission.Metadata, labe
 	if err != nil {
 		return err
 	}
+
+	rs, err := gp.kubernetesClient.Extensions().ReplicaSets(gp.namespace).List(api.ListOptions{
+		LabelSelector: labels.Set(funcLabels).AsSelector(),
+	})
+	if err != nil || len(rs.Items) == 0 {
+		fmt.Println("replicasets is nil or empty")
+		return nil
+	}
+
+	pod.Labels["pod-template-hash"] = rs.Items[0].Labels["pod-template-hash"]
```

Comment:
Relabeling the pod so that it can be adopted by the new replica set could be problematic.
There is a chance that the new replica set set is not created when we relabel the pod.
Async listing the new replica set several times may help. Need to find a better way to transfer ownership of pod.